### PR TITLE
Reindex work after visibility change

### PR DESCRIPTION
**Pull request by laritakr**
Resolves https://github.com/samvera/hyrax/issues/2777

Adding an update_index after a visibility update to the work.
Nested reindexing causes the visibility to not be indexed on the work because the work indexing happens before the visibility is updated in fedora. (see https://github.com/samvera/hyrax/issues/1826 for more info regarding situation)

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by laritakr** [comment]
@cjcolvar Test is now added, and I confirmed that it failed before change and passes after.

**Comment by cjcolvar** [comment]
@elrayle I moved the update_index call into the actor stack where I think it is more natural and useful in more cases beyond the works controller.  If this change doesn't make sense then we can revert it.

File: app/actors/hyrax/actors/interpret_visibility_actor.rb [hunk 1]
```diff
@@ -227,6 +227,13 @@ def apply_embargo(env, intention)
           return unless env.curation_concern.embargo
           env.curation_concern.embargo.save # see https://github.com/samvera/hydra-head/issues/226
         end
+
+        def reindex_visibility_change(env)
+          return true unless env.curation_concern.visibility_changed?
+          env.curation_concern.update_index
+          # Don't try to handle the indexing response and always return true
+          true
```
Comment (elrayle) [inline app/actors/hyrax/actors/interpret_visibility_actor.rb]: Should this be a TODO where we handle the indexing response in the future?  If not, can your comment include why it is find to ignore the indexing response?

Comment (cjcolvar) [reply app/actors/hyrax/actors/interpret_visibility_actor.rb]: I'm concerned that reaching into the response hash (`response["responseHeader"]["status"]`) will be too brittle but I could put that in.  For what it's worth, it looks like even ActiveFedora ignores the rsolr response: https://github.com/samvera/active_fedora/blob/master/lib/active_fedora/indexing.rb#L74-L79

Comment (elrayle) [reply app/actors/hyrax/actors/interpret_visibility_actor.rb]: OK

**Pull request merged**